src/core/drive/page_renderer.ts

```diff
@@ -18,7 +16,7 @@ export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
   }
 
   get shouldRender() {
-    return this.newSnapshot.isVisitable && this.trackedElementsAreIdentical && this.bodyElementMatches
```

Comment:
@agrobbin could you also share some context around this change and the `{ reason: "body_element_mismatch" }` below? If they're crucial, we should include coverage for them.

Reply:
This was done at the request of @dhh in https://github.com/hotwired/turbo/pull/627#issuecomment-1188307421, to trigger a full page reload when there is a mismatch of the configured body element, and provide a `reason` for the reload that is passed along through the invalidation code path, and eventually to the `turbo:reload` event `detail` [here](https://github.com/hotwired/turbo/blob/eda47178e57a5f666c268c54ab459a482faf020b/src/core/native/browser_adapter.ts#L124).

Reply:
Could you outline a scenario where that might occur? Having a test case to cover that code would guard against future regressions.

Reply:
Imagine these 2 HTML documents (abbreviated to convey the intent):

document 1:

```html
<html>
  <head>
    <meta name="turbo-body" content="app">
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>
```

document 2:

```html
<html>
  <head>
    <meta name="turbo-body" content="application">
  </head>
  <body>
    <div id="application"></div>
  </body>
</html>
```

Transitioning between these 2 documents (in either direction) via Turbo will hit this scenario, since the `turbo-body` configurations do not match (`app` vs. `application`).

Reply:
I've added test coverage to exercise these lines. Thank you for sharing an example.